x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/EntSearchPipeline.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+/**
+ * The default Enterprise Search pipeline used by connectors and crawlers
+ */
+public class EntSearchPipeline {
+    @SuppressWarnings("checkstyle:LineLength")
+    private static final String pipelineDefinition =
```

Comment:
Not a leading question, just actual curiosity. Is it idiomatic in the elasticsearch codebase to store these kinds of resources as string literals, rather than as JSON resource files that are read from disk?

Reply:
I couldn't find many examples of storing these kinds of resources as JSON files read from disk--the JSON files I could find were all either test data, or API documentation.

Reply:
I expect it can land in main anytime soon: https://github.com/afoucret/elasticsearch/pull/3
I will open a PR in the main repo once the main event ingest branch will have been merged (hopefully soon).